hibernate-core/src/main/java/org/hibernate/metamodel/mapping/AssociationKey.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or http://www.gnu.org/licenses/lgpl-2.1.html
+ */
+package org.hibernate.metamodel.mapping;
+
+import java.util.List;
+
+/**
+ * @author Andrea Boriero
+ */
+public class AssociationKey {
+	private final String table;
+	private final List<String> columns;
+
+	public AssociationKey(String table, List<String> columns) {
+		this.table = table;
+		this.columns = columns;
+	}
+
+	@Override
+	public boolean equals(Object o) {
+		if ( this == o ) {
+			return true;
+		}
+		if ( o == null || getClass() != o.getClass() ) {
+			return false;
+		}
+
+		final AssociationKey that = (AssociationKey) o;
+		return table.equals( that.table ) && columns.equals( that.columns );
+	}
+
+	@Override
+	public int hashCode() {
+		return table.hashCode();
+	}
```

Comment:
why not include `columns' into hashCode as well? It has been used in 'equals'. Keep from using it doesn't violate the consistency between `hashCode` and `equals`, but it is a little bit confusing.